lib/protocol/hpack/decompressor.rb

```diff
@@ -98,30 +101,56 @@ def read_header
 				pattern = peek_byte
 
 				header = {}
-				header[:type], type = HEADER_REPRESENTATION.find do |_t, desc|
-					mask = (pattern >> desc[:prefix]) << desc[:prefix]
-					mask == desc[:pattern]
-				end
 
-				raise CompressionError unless header[:type]
+				type = nil
+
+
+				case (pattern & MASK_SHIFT_4)
+				when 0x00
+					header[:type] = :no_index
+					type = NO_INDEX_TYPE
+				when 0x10
+					header[:type] = :never_indexed
+					type = NEVER_INDEXED_TYPE
+				# checking if (pattern >> 5) << 5 == 0x20
+				# Since we cleared bottom 4 bits, the 5th
+				# bit can be either 0 or 1, so check both
+				# cases.
+				when 0x20, 0x30
+					header[:type] = :change_table_size
+					type = CHANGE_TABLE_SIZE_TYPE
+				# checking if (pattern >> 6) << 6 == 0x40
+				# Same logic as above, but now over the 4
+				# possible combinations of 2 bits (5th, 6th)
+				when 0x40, 0x50, 0x60, 0x70
+					header[:type] = :incremental
+					type = INCREMENTAL_TYPE
+				# checking if (pattern >> 7) << 7 == 0x80
+				when 0x80, 0x90, 0xa0, 0xb0, 0xc0, 0xd0, 0xe0, 0xf0
+					header[:type] = :indexed
+					type = INDEXED_TYPE
+				else
+					raise CompressionError
```

Comment:
We are missing coverage for this line - can you add a test for the failure case?

Reply:
Never mind, this failure case is impossible (see comment)